### PR TITLE
update testing documentation

### PR DESCRIPTION
**Pull request by josh-m-sharpe**
Addresses #206

### CODE REVIEW

File: docs/guide/testing.md [hunk 1]
```diff
@@ -50,7 +63,8 @@ describe('ExplicitReferenceComponent', () => {
 
 ## Test components using [icon library](../usage/icon-library.md)
 
-When it comes to testing components using icon library you'll need to setup the icon library in one way or another otherwise component will throw because of the missing icon definition. The icon library setup is needed because most likely `AppModule` where you normally add icons to the library is not part of the tested component spec and therefore icons added there are not available.
+When it comes to testing components using icon library you'll need to setup the icon library in one way or another otherwise component will throw because of the missing icon definition.
+The icon library setup is needed because most likely `AppModule` where you normally add icons to the library is not part of the tested component spec and therefore icons added there are not available.
 
 There are two main options how to deal with the icon library in the tests:
```
Comment (devoto13) [inline docs/guide/testing.md]: Please add the example as another section (similar to those below). Currently this PR changes example code provided for a completely different way of using `angular-fontawesome` (explicit reference) and this is not right.

File: docs/guide/testing.md [hunk 2]
```diff
@@ -7,30 +7,43 @@ This guide explains how to test components, which use Font Awesome icons.
...
-@Component({
-  selector: 'app-explicit-reference',
-  template: '<fa-icon [icon]="faUser"></fa-icon>',
+import { FontAwesomeModule, FaIconLibrary, FaConfig } from '@fortawesome/angular-fontawesome';
+import { faStar } from '@fortawesome/free-solid-svg-icons';
+import { faFacebook } from '@fortawesome/free-brands-svg-icons';
+
+@NgModule({
+  declarations: [],
+  providers: [FaIconLibrary, FaConfig],
```
Comment (devoto13) [inline docs/guide/testing.md]: I don't think `providers` and `declarations` are needed here.

File: docs/guide/testing.md [hunk 3]
```diff
@@ -7,30 +7,43 @@ This guide explains how to test components, which use Font Awesome icons.
...
 
 describe('ExplicitReferenceComponent', () => {
   let component: ExplicitReferenceComponent;
   let fixture: ComponentFixture<ExplicitReferenceComponent>;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [FontAwesomeModule], // <--
+      imports: [
+          FontAwesomeModule       // <--
```
Comment (devoto13) [inline docs/guide/testing.md]: I don't think `FontAwesomeModule` needs to be imported as it is added to `exports` of the `FontAwesomeIconsModule`.

**Comment by GanduAnuhya** [comment]
kk

**Comment by devoto13** [comment]
Thanks again for the PR. I've added this documentation in https://github.com/FortAwesome/angular-fontawesome/pull/224.

**Pull request closed without merging**